File: dart/difference-of-squares/lib/difference_of_squares.dart
import 'dart:math';

class DifferenceOfSquares {
  int squareOfSum(int n) {
    var sum = Iterable<int>.generate(n, (x) => x + 1)
        .reduce((value, element) => value + element);
    return pow(sum, 2) as int;
  }

  int sumOfSquares(int n) {
    return Iterable<int>.generate(n, (x) => x + 1)
        .map((x) => pow(x, 2) as int)
        .reduce((value, element) => value + element);
  }

  int differenceOfSquares(int n) {
    return squareOfSum(n) - sumOfSquares(n);
  }
}

File: python/difference-of-squares/difference_of_squares.py
def _get_first_n_natural_numbers(n: int):
    return range(1, n + 1)


def square_of_sum(n: int) -> int:
    return sum(_get_first_n_natural_numbers(n)) ** 2


def sum_of_squares(n: int) -> int:
    return sum(map(lambda n: n ** 2, _get_first_n_natural_numbers(n)))


def difference_of_squares(n: int) -> int:
    return square_of_sum(n) - sum_of_squares(n)
